jsmpp/src/main/java/org/jsmpp/DefaultPDUReader.java

```
@@ -30,27 +32,47 @@
  * 
  */
 public class DefaultPDUReader implements PDUReader {
+	private static final Logger LOG = LoggerFactory.getLogger(DefaultPDUReader.class);
 
     /* (non-Javadoc)
      * @see org.jsmpp.PDUReader#readPDUHeader(java.io.DataInputStream)
      */
     public Command readPDUHeader(DataInputStream in)
             throws InvalidCommandLengthException, IOException {
-        Command header = new Command();
-        header.setCommandLength(in.readInt());
+		final String DEBUG_STR = ":readPDUHeader: ";
+		int available = in.available();
+		if (available == 0 ) {
+			try {
+				Thread.sleep(500);
+			} catch (InterruptedException e) {
+				LOG.error("{} ", DEBUG_STR, e);
+			}
+			return null;
```

What happen if it return null?
I didn't remember it can return null, returning null may break the other? Have you check this?
Or maybe throw an exception would be better?